src/main/java/de/siphalor/amecs/api/KeyModifiers.java

```diff
@@ -1,37 +1,67 @@
 package de.siphalor.amecs.api;
 
+import java.util.Arrays;
+import java.util.BitSet;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.jetbrains.annotations.ApiStatus;
+
 import de.siphalor.amecs.impl.AmecsAPI;
 import de.siphalor.amecs.impl.duck.IKeyBinding;
 import net.fabricmc.api.EnvType;
 import net.fabricmc.api.Environment;
 import net.minecraft.client.option.KeyBinding;
-import java.util.Arrays;
-import java.util.BitSet;
-import java.util.stream.Collectors;
+import net.minecraft.client.util.InputUtil;
 
 /**
  * Defines modifiers for a key binding
  */
 @SuppressWarnings({"WeakerAccess", "UnusedReturnValue"})
 @Environment(EnvType.CLIENT)
 public class KeyModifiers {
-	private BitSet value;
+	/**
+	 * This field is for comparison ONLY. <p>
+	 * Trying to change the modifiers of it will fail with an UnsupportedOperationException
+	 */
+	public static final KeyModifiers NO_MODIFIERS = new KeyModifiersNONE();
+	
+	private static class KeyModifiersNONE extends KeyModifiers {
+		private static final String EXCEPTION_MESSAGE = "You must not alter this Modifiers object";
+		@Override
+		public KeyModifiers setValue(boolean[] value) {
+			throw new UnsupportedOperationException(EXCEPTION_MESSAGE);
+		}
+		@Override
+		public void set(KeyModifier keyModifier, boolean value) {
+			throw new UnsupportedOperationException(EXCEPTION_MESSAGE);
+		}
+		@Override
+		public void unset() {
+			throw new UnsupportedOperationException(EXCEPTION_MESSAGE);
+		}
+	}
+	
+	//using a boolean array here because it is faster and needs less space
+	private final boolean[] value;
 
 	/**
 	 * Constructs new object with no modifiers set
 	 */
 	public KeyModifiers() {
-		this(new BitSet(KeyModifier.getModifierCount()));
+		this(new boolean[KeyModifier.getModifierCount()]);
 	}
 
 	/**
+	 * FOR INTERNAL USE ONLY <p>
```

Comment:
> Also, another weird paragraph tag that doesn't get closed anywhere
As I explained in the other conversation: There is no `</p>` in HTML.

> Why is this in caps?
Because I wanted to be read. Alternatively we could use <h1> ?? We take what you like best

Reply:
> As I explained in the other conversation: There is no </p> in HTML.

This is just wrong. It doesn't make a difference in modern browsers because they will interpret it the same, but `</p>` is valid HTML. Look it up in the specification if you don't believe me.

Reply:
Yes there is a `</p>` you are right. I mixed this with `</br>`. I should change the `<p>` to `<br><br>`